### ISSUE TITLE
SPGAN training problem

### ISSUE THREAD

**Issue by banxia1994** [Simon4Yan/Learning-via-Translation#14]
follow the instructions，I train the SPGAN， but Every time I get stuck in training for the first epoch。
#######
2018-11-07 21:52:58.269039: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
 [*] Loading checkpoint...
None
 [*] No suitable checkpoint!
================ Begining Training time (Wed Nov  7 21:54:31 2018) ================

-------------------------------------------------------
Epoch: (  0) (    1/   50)
######################

It was repeated many times, once to the second epoch，But it's stuck again

################

2018-11-07 21:52:58.268951: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2018-11-07 21:52:58.269018: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2018-11-07 21:52:58.269039: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
 [*] Loading checkpoint...
None
 [*] No suitable checkpoint!
================ Begining Training time (Wed Nov  7 21:54:31 2018) ================

-------------------------------------------------------
Epoch: (  0) (    1/   50)
-------------------------------------------------------
Epoch: (  0) (    6/   50)
-------------------------------------------------------
Epoch: (  0) (   11/   50)
-------------------------------------------------------
Epoch: (  0) (   16/   50)
-------------------------------------------------------
Epoch: (  0) (   21/   50)
-------------------------------------------------------
Epoch: (  0) (   26/   50)
-------------------------------------------------------
Epoch: (  0) (   31/   50)
-------------------------------------------------------
Epoch: (  0) (   36/   50)
-------------------------------------------------------
Epoch: (  0) (   41/   50)
-------------------------------------------------------
Epoch: (  0) (   46/   50)
================ Training time (Wed Nov  7 22:35:27 2018) ================

Model saved in file: ./checkpoints/market2duke_spgan/Epoch_(0)_(50of50).ckpt
-------------------------------------------------------
Epoch: (  1) (    1/   50)

################

Anybody have the same problem？

**Comment by Simon4Yan** [comment]
Hello, I am not an expert at TF. Maybe you could see https://www.cnblogs.com/gaoss/p/7694682.html and https://github.com/tensorflow/tensorflow/issues/7778 for the solution. Good luck to you.

BTW, I will release the pytorch vision code in few weeks.

**Comment by GuangxingHan** [comment]
@Simon4Yan  So is your pytorch code based on this project  https://github.com/layumi/Person_reID_baseline_pytorch? So could you share with us your new experimental results in advance?

**Comment by Simon4Yan** [comment]
Now, SPGAN and feature leanring (IDE+) are based on Pytorch.
We conduct the experiment base on pytorch framework, and the results are higher than our paper. For example, we gain 44.2% in rank accuracy on Duke. With LMP, we get about 48.6%.

You can send E-mail to me, and tell me Wechat. And I will seed our new code to you in advance. Thanks for your attention.

**Comment by banxia1994** [comment]
> Hello, I am not an expert at TF. Maybe you could see https://www.cnblogs.com/gaoss/p/7694682.html and [tensorflow/tensorflow#7778](https://github.com/tensorflow/tensorflow/issues/7778) for the solution. Good luck to you.
> 
> BTW, I will release the pytorch vision code in few weeks.
thx for your reply， i will try for it.

**Comment by xxx2974** [comment]
@Simon4Yan ,hello,do you have the pytorch version code? I can not run this version successfully.

**Comment by JerryZhuzq** [comment]
Hello. In train_spgan_demo.py  I set the gpu_id = 0 which should use my GPU, however, the task manager showed that the algorithm only used CPU, have you went through similar problem?
BTW, could you share with us your pyTorch version, thank you!

**Comment by 2110317008** [comment]
@Simon4Yan hello,do you have the pytorch version code? I can not run this version successfully.If so, I hope you can provide it, I will be very grateful!
